src/integration/java/com/smartcar/sdk/VehicleTest.java

```diff
@@ -69,6 +69,16 @@ public void testImperialHeaders() throws SmartcarException {
         Assert.assertEquals(response.getUnitSystem(), "imperial");
     }
 
+    /**
+     * Tests that the vehicle correctly handles imperial headers.
+     */
+    @Test(groups = "vehicle")
+    public void testRequestIdHeader() throws SmartcarException {
+        SmartcarResponse response = this.vehicle.odometer();
+        // Request ID is a UUID (36 characters)
+        Assert.assertEquals(response.getRequestId().length(), 36);
```

Comment:
thoughts on doing `.setRequestId` and then `getRequestId` ?

Reply:
but if this is an integration test wouldn't we want to see the `requestId` being populated from the actual request?

Reply:
ah yeah, that's fair